backend/human.js

```diff
@@ -131,6 +132,10 @@ module.exports = class extends Player {
     else
       this.sendPack(next);
 
+    if (this.isDecadent) {
+      pullAll(pack, pack);
```

Comment:
I'm not a big fan of having conditions on the game mode, especially if we want to prepare for the winston draft which will be also different. I would maybe use a delegate for the "pick" function or at least a part of it. It can be done later in a refactor if we implement the winston draft. 

I think the simplest way to empty an array is just "pack.length = 0" instead of a "pullAll" but I can be wrong.

Reply:
Agreed, this is inelegant for sure. I would prefer to refactor this before merging the PR rather than letting it wait.

> I would maybe use a delegate for the "pick" function or at least a part of it.

Could you elaborate on what you mean by this? I'm not sure I understand what you mean by this, could you spell it out a bit more for me? :confused:

Reply:
I am referring to the [delegation pattern](https://en.wikipedia.org/wiki/Delegation_pattern). Instead of using a boolean and handle the different strategies inside the pick function of a player, which is not extendable (as you would need to always add another boolean for another mode, and create a huge function pick with a lot of conditional expressions), you can pass to the player object a delegate (pickDelegate or something like that) that will take care of the pick function. Doing so, you move out of the player pick function the true logic and you encapsulate it to other functions. We could make a "regularPickDelegate" which does what is done right now and make a new "decadentPickDelegate" which will take care of the new pick strategy. 

Doing so, your player object will always take one extra argument, a pick delegate, even when we'll implement new pick strategies. The game though needs to choose the right delegate to associate to the player, but it's fine as it knows what type of game it is. 

Instead of delegation, we could also use inheritance (create a DecadentPlayer - I love the name xD) or create a DecadentGame - I would prefer the latter as I feel that it makes more sense to have one class per type of game, as some game mods don't only affect the pick of a player but also how to create a pool, how to share the pool to players etc (I am thinking of the winston draft pool which is a randomization of X boosters, and the rochester draft which reveals the booster to all the players at the same time even though only one player picks a card.

But these considerations are maybe a bit far fetched. I mean that it may be a big deal to implement a delegate or inheritance, I don't know how you feel about it. So I'm ok with keeping the conditionals, or if you want I can try to implement a delegate to your PR or to another branch to show you what I have in mind.

Reply:
@HerveH44 do we really need to go so fancy on that?
Instead, what I would propose is to observe the fact that whilst picking is a player's action, it is still the same for every player in every conceivable draft format, so why don't we just make pick function family static and go for an approach like this:

```
const Human = class {
  constructor (pick_fn, game_type = "Solitaire") {
    this.pick = pick_fn;
    this.game_type = game_type;
  }
  static solitarie_pick() {
    return ("I like it when the cards go psh-psh-psh in " + this.game_type);
  }
  static magic_pick() {
    return (this.game_type + ": the Gathering");
  }
};

const game = {
  player1: new Human(Human.magic_pick, "Magic"),
};

console.log(game.player1.pick())
// expected output: "Magic: the Gathering"
```

It will add another constructor argument to the Human class, but it will keep different pick strategies completely decoupled and it's up to the Game class to select the appropriate function.

Reply:
At least that's what I'm gonna be doing with my Glimpse Draft implementation unless you people will tell me otherwise :)

Reply:
Thanks for the information, that was quite helpful!

For now, I've added a `pickDelegate` constructor param to `Human` and removed the `isDecadent` flag from `Player`. https://github.com/dr4fters/dr4ft/pull/990/commits/f9bef937b1889b7f1446a2e342730fe11ca4ef71. I didn't add this to `Bot` because I've actually disabled bots (and player shuffling) for decadent draft in https://github.com/dr4fters/dr4ft/pull/990/commits/f9c3eb58518d384725070526bffdf320c411811f.

As you mentioned, there's the potential for a considerable refactor here. While I want to avoid putting in kludgy stuff like what was there before (the `isDecadent` flag strewn everywhere), I also want to avoid scope creep in this PR. There's undoubtedly a cleaner way to do this (looking at you, @serokellcao, and your commit https://github.com/serokellcao/dr4ft/commit/cc4bacecee8dd6f77cdebcdbab5f3d30686238fd) but IMHO this PR is becoming enough on its own. Please let me know if you disagree.

Reply:
Yeah, it's fine, let's get yours to get merged first, and then figure out what to do with the Human's polymorphism in pick function.